### DISCUSSION TITLE
ZkSync API issue for blockNumber return

### DISCUSSION THREAD

**Discussion by yaanakbr** [zkSync-Community-Hub/zksync-developers#66] [General]
### Environment

Testnet

### Provide a brief description of the functionality you're trying to implement.

Hey there :) I am using zkSync API to return the latest block, if I use:

https://goerli-api.zksync.io/api/v0.2/blocks?from=latest&limit=1&direction=older

It returns: 

```
{"request":{"network":"goerli","apiVersion":"v02","resource":"/api/v0.2/blocks","args":{"direction":"older","from":"latest","limit":"1"},"timestamp":"2023-07-15T14:23:23.171251873Z"},"status":"success","error":null,"result":{"list":[{"blockNumber":90613,"newStateRoot":"302d877f443608c27be63f9aade9a92c4ab313634c542d2f01b49dc13ce0ad5b","blockSize":26,"commitTxHash":"0xe12648203afcd3a08a88395c351ddfe3c23af82ae7fe001823bba530a6fd463a","verifyTxHash":"0x5bd281ab0c6e24d1f0571211b697181b3a63ccd38168f0db25c2f755a1c7a9ad","committedAt":"2023-07-15T13:56:12.160686Z","finalizedAt":"2023-07-15T13:59:32.164192Z","status":"finalized"}],"pagination":{"from":90613,"limit":1,"direction":"older","count":90613}}}
```

But if you look on the block explorer, the latest `blockNumber` is currently:

https://goerli.explorer.zksync.io/block/9699807

How do I input the API to return `blockNumber` to give me 9699807

Sorry if submitting an issue was wrong by the way. I didn't know if this was expected behavior but seems wrong to me. 

Thank you for any help/feedback in advance!

### What is the specific issue or error you're encountering?

looking for ``BlockNumber`` being returned properly

### Can you share the error messages or logs you're receiving, if any?

N/A

### Have you made any recent changes to the contract before encountering this issue?

N/A

### Are there any external libraries or contracts that your contract interacts with?

N/A

### Can you provide the relevant portions of your contract code where the issue is occurring?

N/A

### Have you tried to isolate the problem, and if so, what were the results?

N/A

### What steps have you already taken to try to resolve the issue?

Trying to read the docs here and follow the "latest" and "lastCommitted" instructions

https://docs.zksync.io/apiv02-docs/#blocks-api-v0.2-blocks-get

**Comment by MexicanAce** [comment]
I believe that you're using the older API that's using zkSync Lite instead of zkSync Era. Try the following intead:

```
curl -X POST -H "Content-Type: application/json" \
--data '{"jsonrpc": "2.0", "id": 1, "method": "eth_blockNumber", "params": [  ]}' \
"https://testnet.era.zksync.dev"
```

That should return:
`{"jsonrpc":"2.0","result":"0x9653e4","id":1}`

And `0x9653e4` is hex for `9851876`

**Comment by yaanakbr** [comment]
Hey @MexicanAce thank you for this! Is there any documents that can support this? I was looking at the V2 version so I thought it was the newest.

**Reply by MexicanAce** [reply]
https://era.zksync.io/docs/

**Comment by yaanakbr** [comment]
Also @MexicanAce just to confirm I can do this via HTTP API and not only RPC, correct? I specifically need the HTTP API call.

**Reply by MexicanAce** [reply]
Unsure exactly what you mean, but this HTTPS POST request is all that exists right now

**Comment by BritvaBo** [comment]
So many questions...
